c++ training/ArraysVectors/Arrays-Vectors/ArraysVectors/DynamicArray.h

```diff
@@ -0,0 +1,241 @@
+		for (int i = 0; i < Size; i++)
+		{
+			NewData[i] = Data[i];
+		}
+
+		Capacity = NewCapacity;
+
+		delete[] Data;
+		Data = NewData;
+		NewData = nullptr;
+	}
+
+public:
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	TDynamicArray() {}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	TDynamicArray(const TDynamicArray<T>& Other)
+	{
+		Append(Other);
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	~TDynamicArray()
+	{
+		delete[] Data;
+		Data = nullptr;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& operator[](const int Index)
+	{
+		return Data[Index];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& operator[](const int Index) const
+	{
+		return Data[Index];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& operator=(const TDynamicArray<T>& Other)
+	{
+		Clear();
+		Append(Other);
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& Front()
+	{
+		return Data[0];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& Front() const
+	{
+		return Data[0];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& Back()
+	{
+		return Data[Size - 1];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& Back() const
+	{
+		return Data[Size - 1];
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T* GetData()
+	{
+		return &Data[0];
```

Comment:
This works nicely, but you can return Data directly. No need to change anything, just commenting in case you're not entirely aware.